src/main/java/org/jvnet/hudson/test/WarExploder.java

```diff
@@ -107,9 +124,12 @@ private static File explode() throws Exception {
             timestamp.setLastModified(war.lastModified());
             new File(explodeDir + ".exploding").delete();
         } else {
-            System.out.println("Picking up existing exploded jenkins.war at "+explodeDir.getAbsolutePath());
+            LOGGER.log(Level.INFO, "Picking up existing exploded jenkins.war at {0}", explodeDir.getAbsolutePath());
         }
 
         return explodeDir;
     }
+
+    private WarExploder() {}
```

Comment:
Unrelated, it may probably cause issues in some custom implementations which instantiate War Exploder due to whatever reason.

I suggest reverting it before merging

Reply:
It consists only of a `static` method. There is no reason to ever instantiate this class. That is the purpose of the `private` constructor—to clarify this.

Anyway we do not guarantee binary compatibility for test harnesses. Upgrades are done only by developers.